compiler-core/src/pretty.rs

```diff
@@ -315,4 +324,20 @@ impl<'a> Document<'a> {
         fmt(writer, limit, 0, docs)?;
         Ok(())
     }
+
+    /// Returns true when the document contains no printable characters
+    /// (whitespace and newlines are considered printable characters).
+    pub fn is_empty(&self) -> bool {
+        use Document::*;
+        match self {
+            Line(n) => *n == 0,
+            ForceBreak => true,
+            String(s) => s.is_empty(),
+            Str(s) => s.is_empty(),
+            // assuming `broken` and `unbroken` are equivalent
+            Break { broken, .. } => broken.is_empty(),
+            FlexBreak(d) | Nest(_, d) | NestCurrent(d) | Group(d) => d.is_empty(),
+            Vec(docs) => docs.iter().all(|d| d.is_empty()),
```

Comment:
I think this will return false for an empty ven when we want it to be true. Prob worth having a test for that